pennylane_lightning/core/lightning_base.py

```diff
@@ -306,18 +307,69 @@ def _process_jacobian_tape(
             "obs_idx_offsets": obs_idx_offsets,
         }
 
+    @staticmethod
+    def _assert_adjdiff_no_projectors(observable):
```

Comment:
Neat. This could be used to recursively check for other unsupported observables.